### PR TITLE
.github/workflows: use latest stable cilium-cli release

### PR DESCRIPTION
**Pull request by tklauser**
Use the `stable.txt` file in the cilium-cli repo to always fetch the
latest stable release.

### CODE REVIEW

File: .github/workflows/conformance-aks.yaml [hunk 1]
```diff
@@ -152,7 +152,7 @@ jobs:
 
       - name: Install Cilium CLI
         run: |
-          curl -LO https://github.com/cilium/cilium-cli/releases/download/v0.8.4/cilium-linux-amd64.tar.gz
+          curl -LO https://github.com/cilium/cilium-cli/releases/download/v0.8.5/cilium-linux-amd64.tar.gz
           sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/bin
```
Comment (rolinh) [inline .github/workflows/conformance-aks.yaml]: Not directly related to this PR but I have 2 questions/suggestions:

- Why is the checksum not being used to verify the tarball?
- If we added a `stable.txt` file to the cilium-cli repo similarly to what is done in the Hubble and Cilium repos, we wouldn't have to open PRs to update Cilium CLI every time a new release is tagged.

TLDR, I'm suggesting to do something like this:
```
export CILUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/download/$CILUM_CLI_VERSION/cilium-linux-amd64.tar.gz{,.sha256sum}
sha256sum --check cilium-linux-amd64.tar.gz.sha256sum
sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/local/bin
rm cilium-linux-amd64.tar.gz{,.sha256sum}
```

Comment (tklauser) [reply .github/workflows/conformance-aks.yaml]: Good points. Let's change it in this PR already. I'll send a PR adding `stable.txt` to `cilium-cli` and will then change this PR accordingly.

Comment (tklauser) [reply .github/workflows/conformance-aks.yaml]: Cilium CLI PR add `stable.txt` (and documenting the release process): cilium/cilium-cli#434

Comment (tklauser) [reply .github/workflows/conformance-aks.yaml]: Updated this PR accordingly.

**Review by rolinh — APPROVED** [review]

**Comment by tklauser** [comment]
All conformance workflows passed:

* AKS https://github.com/cilium/cilium/runs/3084083502?check_suite_focus=true
* AWS-CNI https://github.com/cilium/cilium/runs/3084083368?check_suite_focus=true
* EKS https://github.com/cilium/cilium/runs/3084087176?check_suite_focus=true
* GKE https://github.com/cilium/cilium/runs/3084082814?check_suite_focus=true
* Kind https://github.com/cilium/cilium/runs/3084081122?check_suite_focus=true
* Multicluster https://github.com/cilium/cilium/runs/3084086807?check_suite_focus=true

Removing temporary commit and marking as ready to review.

**Review by pchaigno — APPROVED** [review]
:tada:
